lib/io.c

```diff
@@ -2360,3 +2360,40 @@ grn_pwrite(grn_ctx *ctx, fileinfo *fi, void *buf, size_t count, off_t offset)
 }
 
 #endif /* WIN32 */
+
+bool
+grn_io_warm_path(grn_ctx *ctx, grn_io *io, const char *path)
+{
+  FILE *input = grn_fopen(path, "rb");
+  if (!input) {
+    SERR("[io][warm] failed to open a file: <%s>", path);
+    return false;
+  }
+
+  GRN_LOG(ctx, GRN_LOG_DUMP, "[io][warm] <%s>", path);
+  char buffer[4096];
```

Comment:
No. This is just a temporary buffer size.
There is no more mean of this.

This just reads all data from the given path to put the content of the given path to OS's page cache.

If we name this value, we need to use `BUFFER_SIZE` or something. I think that this is not more meaningful than this. Because `buffer[4096]` means 4096 bytes buffer. And there is only one place that uses the value. We use `sizeof(buffer)` in other place.

So I think that we don't need to name the value.

Reply:
I see, Thank you!